### ISSUE TITLE
ARTE (France) HTTP Error 500 since 2 days

### ISSUE THREAD

**Issue by dnagis** [ytdl-org/youtube-dl#30878]
hello :)

 report: multiple links on ARTE.fr show the same error: I report one below

# youtube-dl --verbose https://www.arte.tv/fr/videos/107231-000-A/bresil-le-retour-de-la-faim/
[debug] System config: []
[debug] User config: []
[debug] Custom config: []
[debug] Command-line args: ['--verbose', 'https://www.arte.tv/fr/videos/107231-000-A/bresil-le-retour-de-la-faim/']
[debug] Encodings: locale UTF-8, fs utf-8, out utf-8, pref UTF-8
[debug] youtube-dl version 2021.12.17
[debug] Python version 3.8.2 (CPython) - Linux-5.6.18-x86_64-with-glibc2.2.5
[debug] exe versions: none
[debug] Proxy map: {}
[ArteTV] 107231-000-A: Downloading JSON metadata
ERROR: Unable to download JSON metadata: HTTP Error 500: Internal Server Error (caused by <HTTPError 500: 'Internal Server Error'>); please report this issue on https://yt-dl.org/bug . Make sure you are using the latest version; see  https://yt-dl.org/update  on how to update. Be sure to call youtube-dl with the --verbose flag and include its complete output.
  File "/usr/lib/python3.8/site-packages/youtube_dl/extractor/common.py", line 634, in _request_webpage
    return self._downloader.urlopen(url_or_request)
  File "/usr/lib/python3.8/site-packages/youtube_dl/YoutubeDL.py", line 2288, in urlopen
    return self._opener.open(req, timeout=self._socket_timeout)
  File "/usr/lib/python3.8/urllib/request.py", line 531, in open
    response = meth(req, response)
  File "/usr/lib/python3.8/urllib/request.py", line 640, in http_response
    response = self.parent.error(
  File "/usr/lib/python3.8/urllib/request.py", line 569, in error
    return self._call_chain(*args)
  File "/usr/lib/python3.8/urllib/request.py", line 502, in _call_chain
    result = func(*args)
  File "/usr/lib/python3.8/urllib/request.py", line 649, in http_error_default
    raise HTTPError(req.full_url, code, msg, hdrs, fp)

**Comment by dirkf** [comment]
Once the above PR is stable, we can back-port it.

**Comment by dirkf** [comment]
However (untested):
https://github.com/yt-dlp/yt-dlp/pull/3302#issuecomment-1105479225

**Comment by dnagis** [comment]
Ok, Arte video download work like a charm with yt-dlp. Thank you very much!

**Comment by dirkf** [comment]
According to the link, yt-dl should be working again too.

**Comment by keimlink** [comment]
Yes, successfully downloaded a video on 2020-04-21 from Arte using youtube-dl. 💯

**Comment by StephaneAscoet** [comment]
> According to the link, yt-dl should be working again too.
> Yes, successfully downloaded a video on 2020-04-21 from Arte using youtube-dl.

What?! How do you do? It doesn't work with me on several computers, several networks... for days and days!
And I can't switch back to Qarte since this one doesn't work anymore for months :-(

**Comment by andreygursky** [comment]
>> be working again too.
>
> What?!

At those moment ARTE activated access over API v1 again. But only temporary. For now one can patch yt-dlp manually using https://github.com/yt-dlp/yt-dlp/pull/3302. This solution uses the new API v2.

**Comment by zebulon-1st** [comment]
Thank you for this patch (API v2). Now it works fine to me too. It was not so easy, especially because I'm neither familiar with python nor with diff. So I applied this patch manually, and yt-dlp works fine with ARTE.tv.

**Comment by oblab** [comment]
> [#30923 (comment)](https://github.com/ytdl-org/youtube-dl/issues/30923#issuecomment-1141829774):
> 
> > I've switched to yt-dlp, which works after replacing the arte.py file with this one.
> > Note that audio and video files apparently need to be downloaded separately after identifying them with the -F swtich (at least based on my so far limited experience). Then you can use ffmpeg to combine them: ffmpeg -i audio.mp4 -i video.mp4 -c:v copy -c:a copy output.mp4.
> 
> Just specify [`-f x+y`](https://github.com/yt-dlp/yt-dlp/pull/3302#issuecomment-1140455036) where x is the video format and y is the audio format.
> 
> Once the yt-dlp PR is [ready](https://github.com/yt-dlp/yt-dlp/pull/3302#issuecomment-1141418769), we can back-port it to yt-dl.

no need to switch to yt-dpl, you can open Developper tools in Firefox, go to network>Media,  copy/paste the link from the mp4 and audio file in youtube-dl command, then use ffmpeg as explain to combine then

**Comment by Atrebate** [comment]
_**no need to switch to yt-dpl**, you can open Developper tools in Firefox, go to network>Media, copy/paste the link from the mp4 and audio file in youtube-dl command, then use ffmpeg as explain to combine then_

Thank you for your solution, I'm very lucky to save Videos again with my 'old' youtube-dl .....!!!!
### 
Un grand merci à OBLAB pour cette solution.
